runtime/src/utxo.rs

```diff
@@ -221,38 +236,25 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
-	/// Redistribute combined reward value evenly among chain authorities
-	fn disperse_reward(authorities: &[H256]) {
-		let reward = <RewardTotal>::take();
-		let share_value: Value = reward
-			.checked_div(authorities.len() as Value)
-			.ok_or("No authorities")
-			.unwrap();
-		if share_value == 0 { return }
-
-		let remainder = reward
-			.checked_sub(share_value * authorities.len() as Value)
-			.ok_or("Sub underflow")
-			.unwrap();
-		<RewardTotal>::put(remainder as Value);
-
-		for authority in authorities {
-			let utxo = TransactionOutput {
-				value: share_value,
-				pubkey: *authority,
-			};
-
-			let hash = BlakeTwo256::hash_of(&(&utxo,
-						<system::Module<T>>::block_number().saturated_into::<u64>()));
-
-			if !<UtxoStore>::contains_key(hash) {
-				<UtxoStore>::insert(hash, utxo);
-				sp_runtime::print("transaction reward sent to");
-				sp_runtime::print(hash.as_fixed_bytes() as &[u8]);
-			} else {
-				sp_runtime::print("transaction reward wasted due to hash collision");
-			}
-		}
+	/// Redistribute combined reward value to block Author
+	fn disperse_reward(author: &Public) {
+		let reward = RewardTotal::take() + T::Issuance::issuance(system::Module::<T>::block_number());
+
+		let utxo = TransactionOutput {
+			value: reward,
+			// TODO Is this the correct way to convert public key into H256?
```

Comment:
should be... i cribbed it from substrate source code

Reply:
actually, you might be able to just do H256::from(public), see how I did it in tests.

Reply:
tried `H256::from` and got
```
error[E0277]: the trait bound `sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::H256: core::convert::From<&sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::sr25519::Public>` is not satisfied
   --> /home/joshy/ProgrammingProjects/utxo-workshop/runtime/src/utxo.rs:247:12
    |
247 |             pubkey: H256::from(author),
    |                     ^^^^^^^^^^ the trait `core::convert::From<&sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::sr25519::Public>` is not implemented for `sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::H256`
    |
```
So I'll leave it as is.